src/firmware/main/math/polynomial_1d.c

```diff
@@ -0,0 +1,66 @@
+#include "firmware/main/math/polynomial_1d.h"
+
+#include <math.h>
+
+/**
+ * Compute the value at the given point for the polynomial formed by the given coeffs
+ * @param coefficients The coefficients that makeup the polynomial, with the highest
+ *                     order coefficients at the start
+ * @param num_coefficients The number of coefficients
+ * @param x The point to get the polynomial value at
+ * @return The polynomial value at the given point
+ */
+float shared_polynomial1d_getValue(const float* coefficients,
+                                   const size_t num_coefficients, float x)
+{
+    float result = 0;
+    for (size_t order = 0; order < num_coefficients; order++)
+    {
+        result += coefficients[num_coefficients - 1 - order] * pow(x, order);
+    }
+    return result;
+}
+
+/**
+ * Calculate the derivative of the polynomial given by the given coefficients
+ *
+ * @param coefficients [in] The coefficients representing the polynomial to differentiate,
+ *                          with the highest order ones first.
+ * @param num_coefficients [in] The number of coefficients
+ * @param derivative_coefficients [out] This will be populated with the coefficients
```

Comment:
Can we doc the expected size of the derivative_coefficient list? Does it make sense to return the size as well?

Reply:
It is already documented here, and I don't think it makes any sense?